abi-versions/v0.2.1/README.md

```diff
@@ -0,0 +1,2166 @@
+# Proxy-Wasm ABI v0.2.1 specification
+
+---
+
+Status: **FINAL** (corrections and/or clarifications are welcomed)
+
+---
+
+# Terminology
+
+- *Plugin* - a WebAssembly module implementing Proxy-Wasm ABI. There are
+  3 types of plugins: *HTTP*, *Stream* (TCP) and *Background task*.
+
+- *Host* - a proxy or another environment that executes Proxy-Wasm Plugin
+  inside the WebAssembly Virtual Machine (WasmVM).
+
+- *Hostcall* - a function call from Plugin to Host.
+
+- *Callback* - a function call from Host to Plugin (usually, as a result
+  of an external event).
+
+- *Plugin Context* (also known as *Root Context*) - a context in which
+  operations not related to any specific request or stream are executed.
+
+- *Stream Context* - a context in which operations from a specific HTTP
+  request/response pair or TCP stream are executed.
+
+
+# Callbacks and Functions
+
+### Callbacks exposed by the Wasm module
+
+All callbacks (entry points named `proxy_on_<event>`) are optional,
+and are going to be called only if exposed by the Wasm module.
```

Comment:
and if the capability is enabled on the host?

Reply:
Technically, capability restrictions are not the part of the ABI surface (although, we should have error codes for capabilities restricting actions specific parameters, if we ever do that).

Nevertheless, I've added some text.